autoapi/_objects.py

```diff
@@ -231,6 +243,22 @@ def _should_skip(self) -> bool:
             self.inherited and "inherited-members" not in self.options
         )
 
+        reason = ""
+        if self.obj.get("hide", False):
+            reason = "marked hidden by mapper"
```

Comment:
This reason is too generic to be helpful. In addition, objects that are hidden this way will be logged twice. Once via a call to `_trace_visibility` where `hide` is originally set, and another in `display`.
Perhaps instead of only storing `obj["hide"] = True`, we should also store a `obj["hide_reason"]` for use by `display`?

Reply:
makes sense!